File: src/1433-Check-If-a-String-Can-Break-Another-String/1433.cpp
class Solution {
public:
    bool checkIfCanBreak(string s1, string s2) {
        int d1[26] = {}, d2[26] = {};
        for (char c : s1) d1[c - 'a']++;
        for (char c : s2) d2[c - 'a']++;
        
        return check(d1, d2) || check(d2, d1);
    }
private:
    bool check(int d1[], int d2[]) {
        int s = 0;
        for (int i = 0; i < 26; i++) {
            s += d1[i] - d2[i];
            if (s < 0) return false;
        }
        return true;
    }
}; 

File: src/1433-Check-If-a-String-Can-Break-Another-String/1433.py
class Solution:
    def checkIfCanBreak(self, s1: str, s2: str) -> bool:
        d1 = collections.Counter(s1)
        d2 = collections.Counter(s2)
        
        def check(a1, a2):
            s = 0
            for c in string.ascii_lowercase:
                s += a1[c] - a2[c]
                if s < 0:
                    return False
            return True
        return check(d1, d2) or check(d2, d1)